packages/smooth_app/lib/pages/folksonomy/folksonomy_provider.dart

```diff
@@ -4,134 +4,43 @@ import 'package:collection/collection.dart';
 import 'package:flutter/material.dart';
 import 'package:openfoodfacts/openfoodfacts.dart';
 import 'package:smooth_app/database/dao_folksonomy.dart';
+import 'package:smooth_app/database/dao_transient_folksonomy.dart';
 import 'package:smooth_app/database/local_database.dart';
-import 'package:smooth_app/pages/product/common/product_refresher.dart';
-import 'package:smooth_app/query/product_query.dart';
+import 'package:smooth_app/pages/folksonomy/folksonomy_manager.dart';
 
 class FolksonomyProvider extends ValueNotifier<FolksonomyState> {
-  FolksonomyProvider(this.barcode, this.localDatabase)
+  FolksonomyProvider(this.barcode, this._localDatabase)
     : super(const FolksonomyStateLoading()) {
-    fetchProductTags();
+    unawaited(_init());
   }
 
   final String barcode;
-  final LocalDatabase localDatabase;
-  String? _bearerToken;
+  final LocalDatabase _localDatabase;
+  late final DaoFolksonomy _daoFolksonomy;
+  late final FolksonomyManager _folksonomyManager;
```

Comment:
We don't need the `_localDatabase` field, and the code would be easier to read if you set both `_dao*` fields in the constructor.
But your code does work.